UefiPayloadPkg/PayloadLoaderPeim/FitPayloadLoaderPeim.c

```diff
@@ -128,6 +272,1018 @@ EFI_PEI_PPI_DESCRIPTOR  gPpiLoadFilePpiList = {
+                                              &(PciData.Hdr.VendorId)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint16,
+                                              PCI_DEVICE_ID_OFFSET,
+                                              sizeof (PciData.Hdr.DeviceId),
+                                              &(PciData.Hdr.DeviceId)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint8,
+                                              PCI_REVISION_ID_OFFSET,
+                                              sizeof (PciData.Hdr.RevisionID),
+                                              &(PciData.Hdr.RevisionID)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint16,
+                                              PCI_SVID_OFFSET,
+                                              sizeof (PciData.Device.SubsystemVendorID),
+                                              &(PciData.Device.SubsystemVendorID)
+                                              );
+
+          Status = mPciDevicePpi->PciIo.Pci.Read (
+                                              &mPciDevicePpi->PciIo,
+                                              (EFI_PCI_IO_PROTOCOL_WIDTH)EfiPciWidthUint16,
+                                              PCI_SID_OFFSET,
+                                              sizeof (PciData.Device.SubsystemID),
+                                              &(PciData.Device.SubsystemID)
+                                              );
+        }
+
+        Data16 = CpuToFdt16 (PciData.Device.SubsystemID);
+        Status = FdtSetProp (Fdt, GmaNode, "subsystem-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Device.SubsystemVendorID);
+        Status = FdtSetProp (Fdt, GmaNode, "subsystem-vendor-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.RevisionID);
+        Status = FdtSetProp (Fdt, GmaNode, "revision-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.DeviceId);
+        Status = FdtSetProp (Fdt, GmaNode, "device-id", &Data16, sizeof (UINT16));
+
+        Data16 = CpuToFdt16 (PciData.Hdr.VendorId);
+        Status = FdtSetProp (Fdt, GmaNode, "vendor-id", &Data16, sizeof (UINT16));
+      }
+
+      if (PciRootBridgeInfo->RootBridge[Index].Bus.Base == PcdGet8 (PcdPchBus)) {

```

Comment:
This is for serial device under Intel eSPI controller.
After moving this to separate library function, we can just hardcode bus number like ESPI_DEV_NUM, ESPI_FUNC_NUM, other platform/soc will implement their onw library function to reflect their hierarchical PCI tree structure.

Reply:
So, I don't know what the input to this driver is: COREBOOT tables, SBL HOBs, UPL data, or something else. If we have "isa" in the FDT (or something describing that), can we use it? Or, if PCI enumeration is done, then can we just pick the device with the "ISA bridge" PCI class?

Reply:
we will consider hob as input for providing such bus/dev/func information for different platforms.

Reply:
Done.